notebooks/Double Machine Learning Examples.ipynb

```diff
@@ -0,0 +1,965 @@
+   "cell_type": "code",
+   "execution_count": 19,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAlsAAAF3CAYAAACSQ46hAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4zLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvnQurowAAIABJREFUeJzs3XlYldXax/HvAjaDAoLgjDggODCK4DyUI6aVomVlg3ZKrWNa1Fs2Ht/ekyfNOTEzpyazk+bJrEycZw3REiVAFBFHBGQeN+v9w+JoJSDuzeT9uS4v2c+w1r0B4ed6nmctpbVGCCGEEEKYh0V1FyCEEEIIUZdJ2BJCCCGEMCMJW0IIIYQQZiRhSwghhBDCjCRsCSGEEEKYkYQtIYQQQggzkrAlhBBCCGFGEraEEEIIIcxIwpYQQgghhBlJ2BJCCCGEMCOr6i7geq6urrp169bVXYYQQgghRLkOHz58RWvdqLzjalTYat26NZGRkdVdhhBCCCFEuZRSZypynFxGFEIIIYQwIwlbQgghhBBmJGFLCCGEEMKMatQ9W3+lqKiI5ORk8vPzq7sUUQG2tra4ublhMBiquxQhhBCiRqjxYSs5ORkHBwdat26NUqq6yxFl0FqTmppKcnIybdq0qe5yhBBCiBqhxl9GzM/Px8XFRYJWLaCUwsXFRUYhhRBCiOvU+LAFSNCqReRrJYQQQtyoVoSt6pacnMz999+Pp6cnHh4eTJ06lcLCQgB27NhBgwYN6Ny5Mx06dOCll14qPW/VqlU0atSIgIAAAgICePzxx2+rjsTERFavXl36OjIykilTptxWm79btWoV58+fL/e4r776Cm9vbywsLGRONCGEEKICJGyVQ2tNaGgoI0aMID4+nri4OLKzs3n99ddLj+nTpw9HjhzhyJEjbNy4kb1795buGzNmDEePHuXo0aN88sknt1XLH8NWUFAQCxcuvK02f1fRsOXj48PXX39N3759TdKvEEIIUddJ2CrHtm3bsLW1Zfz48QBYWloyb948VqxYQW5u7g3H2tnZERAQwLlz5yrcfkpKCqNGjSI4OJjg4ODSoLZz587SEbHOnTuTlZXFtGnT2L17NwEBAcybN48dO3YwfPhwAKZPn84TTzzB4MGDad26NV9//TUvv/wyvr6+hISEUFRUBMDbb79NcHAwPj4+TJgwAa01a9euJTIykrFjxxIQEEBeXh6HDx+mX79+dOnShSFDhnDhwgUAOnbsSPv27W/78yqEEELcKWr804jX+99vj3PifKZJ2+zU3JF/3Ot90/3Hjx+nS5cuN2xzdHTE3d2dkydP3rA9PT2d+Pj4G0Z9vvzyS/bs2QPA1KlTS0Pb76ZOncoLL7xA7969SUpKYsiQIcTExDB79mzCw8Pp1asX2dnZ2Nra8u677zJ79mw2btwIXLuEeb2EhAS2b9/OiRMn6NGjB+vWrWPWrFmMHDmS7777jhEjRjB58mTeeustAB577DE2btzI6NGjWbRoEbNnzyYoKIiioiKee+45vvnmGxo1asSXX37J66+/zooVK27tkyuEEEKI2hW2qoPW+i9v+r5+++7du/Hz8yM2NpZp06bRtGnT0uPGjBnDokWLbtr+li1bOHHiROnrzMxMsrKy6NWrF2FhYYwdO5bQ0FDc3NzKrXXo0KEYDAZ8fX0xGo2EhIQA4OvrS2JiIgDbt29n1qxZ5ObmkpaWhre3N/fee+8N7cTGxhIdHc2gQYMAMBqNNGvWrNz+hRBCiOp2s9/b1alWha2yRqDMxdvbm3Xr1t2wLTMzk7Nnz+Lh4UFqaip9+vRh48aNxMXF0bt3b0aOHElAQECF2i8pKWH//v3Y2dndsH3atGkMGzaM77//nu7du7Nly5Zy27KxsQHAwsICg8FQ+s1mYWFBcXEx+fn5PPvss0RGRtKyZUumT5/+l9M0aK3x9vZm//79FXoPQgghRE2QU5TD37f+nSd9nqSvW825t1ju2SrHgAEDyM3NLb253Wg08uKLLzJu3Djq1at3w7FeXl68+uqrzJw5s8LtDx48+IaRr6NHjwLXLgn6+vryyiuvEBQUxK+//oqDgwNZWVmVfi+/BytXV1eys7NZu3Zt6b7r227fvj0pKSmlYauoqIjjx49Xul8hhBCiKqyMXsnRy0eru4w/kbBVDqUU69ev56uvvsLT0xMvLy9sbW2ZMWPGXx4/adIkdu3axenTpyvU/sKFC4mMjMTPz49OnTqxZMkSAObPn4+Pjw/+/v7Y2dkxdOhQ/Pz8sLKywt/fn3nz5t3ye3FycuLpp5/G19eXESNGEBwcXLpv3LhxTJo0iYCAAIxGI2vXruWVV17B39+fgIAA9u3bB8D69etxc3Nj//79DBs2jCFDhtxyHUIIIYQ5TPSbyLLBy2rUqBaA0lpXdw2lgoKC9B/nboqJiaFjx47VVJGoDPmaCSGEqCpaa1ZEr2Ck50ga2jas0r6VUoe11kHlHScjW0IIIYSotd4/8j7zo+azMWFjdZdyUxK2hBBCCFErfXL8Ez469hGjvUbzWKfHqrucmzJb2FJKtVdKHb3uT6ZS6nlz9SeEEEKIO8f6+PW8F/keg1oN4o1ub9S46R6uZ7apH7TWsUAAgFLKEjgHrDdXf0IIIYS4MxQZi1h5fCW9mvfi3T7vYmlhWd0llamq5tkaACRorc9UUX9CCCGEqKMMlgZWDFlBPat6WFtaV3c55aqqe7YeAr6oor6EEEIIUQdFXYpi+r7pFJUU4WrnSj1DvfJPqgHMHraUUtbAfcBXN9k/QSkVqZSKTElJMXc5lZKcnMz999+Pp6cnHh4eTJ06lcLCQuDa+oQNGjSgc+fOdOjQgZdeeqn
```

Comment:
Adding polynomial featurizer will show more variation with income, but not like Miruna's 'S' shaped curve. In this case since we don't know the true TE, it's hard to say which degree of polynomial features fits best. Another issue here, bootstrap CI works even worse for higher degrees, will dig into it.

Reply:
I wonder if this would still be true even if you add `shuffle=True` to the cross validation...